.github/workflows/e2e_upgrade.yml

```diff
@@ -0,0 +1,138 @@
+name: MatrixOne e2e upgrade CI
+
+on:
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+    branches: [ main,'[0-9]+.[0-9]+.[0-9]+' ]
+
+jobs:
+  multi-cn-bvt-linux-x86:
+    runs-on: ubuntu-latest
+    name: Multi-CN e2e BVT Test on Linux/x64(LAUNCH, PUSH)
+    timeout-minutes: 60
+
+    steps:
+      - name: checkout head
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: '3'
+          path: ./head
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          ref: ${{ github.event.pull_request.head.ref }}
+
+      - name: checkout upstream
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: '3'
+          path: ./upstream
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
```

Comment:
if upstream is `matrixorigin/matrixone`, shouldn't it be `base` here instead of `head`? :)

Reply:
good catch, fixed.